src/mpid/ch4/src/ch4_init.h

```diff
@@ -246,9 +246,8 @@ MPL_STATIC_INLINE_PREFIX int MPID_Init(int *argc,
         }
     }
 #endif
-
-    MPIR_Process.attrs.tag_ub = (1ULL << MPIDI_CH4U_TAG_SHIFT) - 1;
-    /* discuss */
+    /* Set tag_ub to an sane safe value during initialization and set it to the actual value later. */
+    MPIR_Process.attrs.tag_ub = INT_MAX - 1;
```

Comment:
IMO we can just remove this and rely on the netmod hook to set a value. ch4 itself shouldn't set any limit.

Reply:
Yeah, I'm not 100% sure why this code is there to begin with. I thought it was because of something during initialization, but now I'm not so sure. I'll try taking it out.